src/lib.rs

```diff
@@ -524,16 +535,61 @@ pub trait Parser<'a, I: Input<'a>, O, E: ParserExtra<'a, I> = extra::Default>:
     /// # use chumsky::prelude::*;
     /// use std::ops::Range;
     ///
-    /// // It's common for AST nodes to use a wrapper type that allows attaching span information to them
-    /// #[derive(Debug, PartialEq)]
-    /// pub struct Spanned<T>(T, SimpleSpan<usize>);
+    /// # mod intern {
+    /// #     #[derive(Copy, Clone, Debug, PartialEq)]
+    /// #     pub struct Key(usize);
+    /// #     pub struct Interner {
+    /// #         strings: Vec<String>,
+    /// #     }
+    /// #     impl Interner {
+    /// #         pub fn new() -> Interner { Interner { strings: Vec::new() } }
+    /// #         pub fn intern(&mut self, str: &str) -> Key {
+    /// #             let pos = self.strings.iter().position(|s| s == str);
+    /// #             match pos {
+    /// #                 Some(pos) => Key(pos),
+    /// #                 None => {
+    /// #                     self.strings.push(str.to_string());
+    /// #                     Key(self.strings.len() - 1)
+    /// #                 }
+    /// #             }
+    /// #         }
+    /// #         pub fn get(&self, key: Key) -> &str {
+    /// #             &self.strings[key.0]
+    /// #         }
+    /// #     }
+    /// # }
```

Comment:
Good example! I'd even suddenly making the interner itself visible, or spending on `lasso`

Reply:
By 'spending' do you mean pulling it in and actually using it? If you're fine with adding it as a dev dependency, I think that would be good - I'd be hesitant to expose the stub interner here - it's not the *worst* implementation but it's not one I'd advise exactly.

Reply:
Sorry, I think I'd had a bit to drink and was typing on mobile when I wrote this. Should have been:

> Good example! I'd even suggest making the interner itself visible or depending on `lasso` (i.e: using `lasso` is part of the doc example given that it's a commonly used interning crate)